### ISSUE TITLE
Works well with Microsoft Flight Simulator!

### ISSUE THREAD

**Issue by gfxonline** [Nukem9/dlssg-to-fsr3#64]
Hello!

Just wanted to confirm that I tried this with MSFS and it seems to work well so far, there are issues with the in-game HUDs but those can be turned off in the "Assistance options" menu. Otherwise I don't notice any artifacts or strange issues, but I will continue to test this further in the coming days and I will update if I see anything unusual.

Thank you so much for bringing this mod to us, you just doubled my FPS from ~30 to 60 with the same graphics settings!
Cheers!

**Comment by ebisettia** [comment]
Please!!!, how did you do it?, can you please provide the steps to achieve this? videos or images?
I appreciate it. Thank you in advance for your help.
Have a good one!

**Comment by gfxonline** [comment]
> Please!!!, how did you do it?, can you please provide the steps to achieve this? videos or images? I appreciate it. Thank you in advance for your help. Have a good one!

I just followed the steps in the readme file. Copy the two dll files next to the "flightsimulator.exe" file and then run the sim.
You need to be on DX12, turn off VSync, then toggle "Frame generation" to ON. That's pretty much it!

In other news, I also tried it with HDR and it seems to work without problems.
In-game panels (ATC, Weather, etc.) will flicker while FG is on and that's the only issue I've found so far.
Additionally, the sim will be stuttery mess with FG if your base FPS are less than 15, the solution is to lower your TLOD a bit to achieve 20FPS or higher in-sim, and that will get you a smooth 40 FPS with this mod.

**Comment by ForthtuN** [comment]
Yup, HDR works fine. Just follow the "readme" steps. Basically put the files in the directory of msfs2020

**Comment by ebisettia** [comment]
Awesome!, that's what I wanted to know, how to activate it.  I usually have more than 30 fps, hopefully I should be fine. I don't use the in-game panels, well, rarely. oh! wait!, GSX uses them!, hmmm, Well I have to find out myself.  Hopefully, that issue with the in-game panels will be fixed soon. Thanks again!!!

**Comment by ebisettia** [comment]
Bummer, it completely destroyed my FPS in the game. Went half what I have regularly. I selected DLSS, then Quality, which is what I always use, and then I saw the Frame Gen option available, turned it on, and got that awful FPS.  Here are the pics, I don't know if I did something wrong.  I have a 5900X, 32 gigs of RAM, RTX3080TI FTW ULTRA, 2 TB SSD 990 PRO.

![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 10 48 081](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/bde9128d-932a-446a-bd5a-9885d7b6d3fb)
![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 10 36 581](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/5ca9bb82-539d-4f75-9a47-dddce96f02e4)
![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 02 57 641](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/a61247bc-87b9-414f-9aad-0e0149031ea0)
![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 03 06 221](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/0bff9c40-67b5-479c-8db3-14f27b4f31e1)

**Comment by gfxonline** [comment]
> Bummer, it completely destroyed my FPS in the game. Went half what I have regularly. I selected DLSS, then Quality, which is what I always use, and then I saw the Frame Gen option available, turned it on, and got that awful FPS.  Here are the pics, I don't know if I did something wrong.  I have a 5900X, 32 gigs of RAM, RTX3080TI FTW ULTRA, 2 TB SSD 990 PRO.
> 
> ![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 10 48 081](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/bde9128d-932a-446a-bd5a-9885d7b6d3fb)
> ![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 10 36 581](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/5ca9bb82-539d-4f75-9a47-dddce96f02e4)
> ![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 02 57 641](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/a61247bc-87b9-414f-9aad-0e0149031ea0)
> ![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 03 06 221](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/0bff9c40-67b5-479c-8db3-14f27b4f31e1)
> 

Strange!
Have you tried disabling DLSS and keeping FG on? I'm using TAA with 100% resolution scaling and I definitely did get a considerable boost up to 90%.

**Comment by ebisettia** [comment]
This is TAA with FG, still low FPS, and really high MS.
Seems this thing isn't for me, for some reason, I don't know why, it harms my FPS instead of boosting it.

![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 40 08 74](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/0dc2b015-4483-4c0f-ad0a-15b4692cd2b4)
<img width="1920" alt="Microsoft Flight Simulator Screenshot 2023 12 18 - 20 40 27 21" src="https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/9e90f324-285c-4a82-8fb8-c13fed1b75cf">

**Comment by ebisettia** [comment]
PS.- In-game windows do not flicker in my PC, they are normal as seen in the pic.

![Microsoft Flight Simulator Screenshot 2023 12 18 - 20 42 03 23](https://github.com/Nukem9/dlssg-to-fsr3/assets/75552858/1b48b02a-ad20-462e-82af-1e7b88f6c3be)

**Comment by gfxonline** [comment]
In my case, they flicker only when panning the camera, which is a lot when I have head tracking on, but it's within tolerance I'd say. A small price to pay for the doubled FPS so no complaints there :)

**Comment by apolinski** [comment]
> You need to be on DX12,

That was NOT I repeat NOT mentioned anywhere in the instructions.  Would have been nice to know.

**Comment by Igihara** [comment]
> > You need to be on DX12,
> 
> That was NOT I repeat NOT mentioned anywhere in the instructions. Would have been nice to know.

It's well documented that FSR 3 only works on DX12 games

**Comment by gfxonline** [comment]
> > You need to be on DX12,
> 
> That was NOT I repeat NOT mentioned anywhere in the instructions.  Would have been nice to know.

It's a requirement for frame generation to work, otherwise the option to toggle it in won't appear, even for 40xx generation cards without the mod.

It was maybe assumed by the mod authors that people were aware of that.

Hope the mod works well for you!

**Comment by RoooDog** [comment]
I can't get this to work. 
7800x3D/RTX 3080ti on 546.01 HAGS on / DX12. nvdia sig checks disabled
Have installed the two dlls into the content folder as well as cleared out my shader cache and the load-in pop-up or fg option in the ui setting do not show. At a loss to why it's not showing

**Comment by gfxonline** [comment]
I can't see any reason why it wouldn't work for you. I have a super old 7700K with a 3080 and it worked instantly.
I know the dev has removed the popup dialog since version 0.60 but you should at least see the Frame Generation option in "display options".
Did you make sure that the folder where you copied the DLLs to is the same one where the "flightsimulator.exe" resides?

**Comment by Gl3banya** [comment]
DX12. 3080ti, Don't see Frame Gen option in the menu. Tried everything

**Comment by Onkelpuh** [comment]
I have the same problem, the Frame Generation setting in the MSFS options is not displayed in the graphics options even though I did everything according to the instructions

**Comment by milamber182** [comment]
> I have the same problem, the Frame Generation setting in the MSFS options is not displayed in the graphics options even though I did everything according to the instructions

I had to enable DX12 and change DLSSG from 0 to 1 in UserCfg.opt to unlock it the menu.

**Comment by milamber182** [comment]
Testing in flight training tutorial I get about 90fps at 4K Ultra settings without FG. With the mod I can get up to my capped fps of 141. Unfortunately the UI flicker is so bad when moving the camera I had to disable it. I'd love to see it fixed somehow.

**Comment by Onkelpuh** [comment]
> > I have the same problem, the Frame Generation setting in the MSFS options is not displayed in the graphics options even though I did everything according to the instructions
> 
> I had to enable DX12 and change DLSSG from 0 to 1 in UserCfg.opt to unlock it the menu.

where can I find the file ?

**Comment by Onkelpuh** [comment]
> > Ich habe das gleiche Problem, die Einstellung Frame-Generierung in den MSFS-Optionen wird in den Grafikoptionen nicht angezeigt, obwohl ich alles gemäß den Anweisungen gemacht habe
> 
> Ich musste DX12 aktivieren und DLSSG von 0 auf 1 in UserCfg.opt ändern, um es im Menü zu entsperren.

Unfortunately it still doesn't work for me, the setting in the options doesn't appear for me, I have no idea what the problem is but I'll let it go
